### PR TITLE
feat(#2015): Fix concurrency problem with `LogFormatTest`

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Fix concurrency problem with `LogFormatTest` and remove the corresponding puzzle.

Closes: #2015

---

## PR-Codex overview
This PR adds a new dependency and updates the test class `LogFormatTest` to use a `BlockingQueue` instead of an `AtomicReference` to store log events. It also replaces a disabled test with a timeout annotation.

### Detailed summary
- Added a new dependency
- Replaced `AtomicReference` with `BlockingQueue` in `MockAppender`
- Replaced a disabled test with a timeout annotation in `LogFormatTest`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@mximp Could you have a look, please?

File: eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java [hunk 1]
```diff
@@ -99,22 +88,24 @@ private static final class MockAppender extends WriterAppender {
          */
         private MockAppender(final Appender console) {
             this.console = console;
-            this.last = new AtomicReference<>();
+            this.events = new ConcurrentLinkedDeque<>();
```
Comment (mximp) [inline eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java:91]: @volodya-lombrozo What is the purpose of deqeue here? Can queue fit?

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java]: @mximp Yes, of corse. Thank you!

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java]: Done

**Review by mximp — APPROVED** [review]

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you merge that changes, please?

**Comment by volodya-lombrozo** [comment]
Rultor revealed the problem with the code:

```
[ERROR] Failures:
[ERROR]   LogFormatTest.printsFormattedMessage:60
Expected: is <true>
     but: was <false>
[INFO]
[ERROR] Tests run: 269, Failures: 1, Errors: 0, Skipped: 1
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for eo 1.0-SNAPSHOT:
[INFO]
[INFO] eo ................................................. SUCCESS [ 17.624 s]
[INFO] eo-parser .......................................... SUCCESS [ 35.845 s]
[INFO] eo-maven-plugin .................................... FAILURE [01:55 min]
[INFO] eo-runtime ......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
```
Looks like the fix doesn't work.

**Comment by volodya-lombrozo** [comment]
@mximp Could you take a look again please? I've changed the logic little bit in order to catch async log messages.

File: eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java [hunk 2]
```diff
@@ -99,22 +83,49 @@ private static final class MockAppender extends WriterAppender {
...
 
         /**
-         * Formatted log message.
-         *
-         * @return Real log message.
+         * Check if any log message matches the regex.
+         * @param regex The regex to match.
+         * @return True if any log message matches the regex.
+         */
+        private boolean containsMessage(final String regex) {
```
Comment (mximp) [inline eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java:100]: @volodya-lombrozo would `contains` name fit?

File: eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java [hunk 3]
```diff
@@ -99,22 +83,49 @@ private static final class MockAppender extends WriterAppender {
...
+                throw new IllegalStateException(interrupt);
+            }
+        }
+
+        /**
+         * Get the last log message.
+         * @return The last log message.
+         * @throws InterruptedException If interrupted.
+         */
+        private String lastMessage() throws InterruptedException {
```
Comment (mximp) [inline eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java:118]: @volodya-lombrozo would just `last` fit?

**Review by mximp — APPROVED** [review]

File: eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java (hunk 2 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java]: @mximp Done, thanks!

File: eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java (hunk 3 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/LogFormatTest.java]: @mximp Done, thanks!

**Comment by volodya-lombrozo** [comment]
@mximp Could you have a look, again, please?

**Review by mximp — APPROVED** [review]

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you merge that changes, please?

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**